### PR TITLE
fix(client/android): fix android network receiver while still supporting android 10

### PR DESCRIPTION
**Pull request by daniellacosse**
https://github.com/user-attachments/assets/d14b29d9-82c6-41a7-af35-053796a19a99

### CODE REVIEW

File: client/src/cordova/plugin/android/java/org/outline/OutlinePlugin.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ protected void pluginInitialize() {
     IntentFilter broadcastFilter = new IntentFilter();
     broadcastFilter.addAction(VpnTunnelService.STATUS_BROADCAST_KEY);
     broadcastFilter.addCategory(context.getPackageName());
-    context.registerReceiver(vpnTunnelBroadcastReceiver, broadcastFilter, context.RECEIVER_NOT_EXPORTED);
+    context.registerReceiver(vpnTunnelBroadcastReceiver, broadcastFilter, context.RECEIVER_EXPORTED);
```
Comment (fortuna) [inline client/src/cordova/plugin/android/java/org/outline/OutlinePlugin.java:138]: We need explanations on what these changes mean and why they are needed.

File: client/src/cordova/plugin/plugin.xml [hunk 2]
```diff
@@ -39,10 +39,12 @@
       <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
       <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
       <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
+      <uses-permission android:name="android.permission.FOREGROUND_SERVICE_SYSTEM_EXEMPTED" />
```
[the remark this answers is not in the document]
Comment (fortuna) [reply client/src/cordova/plugin/plugin.xml:42]: You need to pass [FOREGROUND_SERVICE_TYPE_SYSTEM_EXEMPTED](https://developer.android.com/reference/android/content/pm/ServiceInfo#FOREGROUND_SERVICE_TYPE_SYSTEM_EXEMPTED) to the startForeground call:
image

**Review by fortuna — COMMENTED** [review]
Please make sure the client is working after the changes

**Review by fortuna — COMMENTED** [review]
Please add comments to the code explaining why we need them

**Comment by sbruens** [comment]
> Please add comments to the code explaining why we need them

Could you still do this? It will be helpful for future readers to know what settings are important and why, so they don't change them.

**Review by sbruens — APPROVED** [review]
See my comment on adding some code comments where appropriate. Otherwise this LGTM, assuming it works on the various Android versions.

**Comment by sbruens** [comment]
> Imagine for example commenting "sets the width to 100%" on the css property `width: 100%` every time (which may actually not always be the case if the element's display property is set to `inline` for instance). Comments like that on every other line ultimately pollute the source with "eventually false" information, imo.

To be fair, that example is not quite the same. I'm not suggesting you comment _what_ the code is doing, but _why_ it's needed (see http://go/comments#reasons).

> I think as a compromise I can simply link to the relevant documentation in the comment, rather than write a paragraph attempting to explain how android works to the reader. The hope there is the documentation stays up to date. Is that fair?

Thanks, yeah that's totally fine. I'm not expecting a liberally commented PR. You're just making very specific changes that are clearly non-trivial, but they are not self-describing. So to avoid future readers needing to dig it will be helpful context on _why_ these specific settings are needed, or what the magic number means, so that someone doesn't inevitable try to remove it, especially because we don't have a regression tests for this.

**Review by sbruens — APPROVED** [review]

**Pull request merged**